### PR TITLE
Move the e2e histograms code to somewhere where it can be exported

### PR DESCRIPTION
**Pull request by zenador**
#### What this PR does

Minor refactor of integration/util.go to separate concerns more cleanly, moving the e2e histograms code to its own package, which will allow us to export it without exporting a lot of other unnecessary code.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

None as these are minor, non-user-facing changes.

### CODE REVIEW

File: integration/e2ehistograms/e2ehistograms.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
```
Comment (narqo) [inline integration/e2ehistograms/e2ehistograms.go:2]: Does this package still need this `go:build` directive?

**Review by narqo — APPROVED** [review]
(nit) On nitpicking side of things, maybe we can name the new package `e2eseries`. I don't think it's worth it to have a dedicated E2E package of a single *.go file per-series type, should we ever add more such helpers.

Works for me otherwise 🔥

File: integration/e2ehistograms/e2ehistograms.go (hunk 1 above)
Comment (zenador) [reply integration/e2ehistograms/e2ehistograms.go]: Good point, removed.

**Comment by zenador** [comment]
Keeping the name for now, because it's quite unlikely that we'll have a new series type anytime soon, and we can easily rename at that point if necessary. Also we keep the e2e generators for floats somewhere else (e2e repo), and we have the histogram generators in this mimir repo because the e2e repo seems to be in use by non-metric products that don't have histograms, so `e2eseries` might be confusing as it doesn't include the float generators.

**Pull request merged**